packages/postgres-database/src/simcore_postgres_database/migration/versions/967cff1d52ac_new_resource_tracker_container_table.py

```diff
@@ -0,0 +1,49 @@
+"""New resource_tracker_container table
+
+Revision ID: 967cff1d52ac
+Revises: c8f072c72adc
+Create Date: 2023-06-13 06:37:31.542190+00:00
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "967cff1d52ac"
+down_revision = "c8f072c72adc"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "resource_tracker_container",
+        sa.Column("id", sa.String(), nullable=False),
```

Comment:
I see this is the primary key.

This would be the container's id from docker I would guess. Now I have a very good question.
What is the change that two containers over the lifespan of the application (years) will be generated with the same id?

If there is a change of collision, maybe also include the image (key version) in the primary key. If you do so, you will need to query via id and image name or just add an index on id to make the query fast.

I would guess this table will be growing quite a lot. We might want to keep this in mind and male the queries go fast at least

Reply:
I guess the chances of collision are really small (secondly & more important is that we will not store this table for a long time) my idea is to have an additional table "aggregated view" that would update a few times throughout the day (something similar to AWS Cost explorer) and therefore afterward we will always delete rows from this table that are already in the aggregated table.